templates/default/datadog.conf.erb

```diff
@@ -15,7 +15,11 @@ autorestart: <%= node['datadog']['autorestart'] %>
 skip_ssl_validation: <%= node['datadog']['web_proxy']['skip_ssl_validation'] %>
 <% end -%>
 
+<% if node['datadog']['tags'].respond_to?(:each_pair) -%>
+tags: <%= node['datadog']['tags'].reject{|k,v| v.empty? }.map{|k,v| "#{k}:#{v}" }.join(',') %>
```

Comment:
Style nitpicks: spacing and unused block arg representation.
Should be `node['datadog']['tags'].reject { |_k, v| v.empty? }.map { |k, v| "#{k}:#{v}" }.join(',')`